x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -510,6 +512,24 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             return SourceValueFetcher.identity(name(), context, format);
         }
 
+        @Override
+        public BlockLoader blockLoader(BlockLoaderContext blContext) {
+            if (blContext.aggregationHint() == null) {
+                throw new UnsupportedOperationException("aggregation hint is required");
+            }
+
+            return switch (blContext.aggregationHint()) {
+                case "max" -> delegateFieldType(Metric.max).blockLoader(blContext);
+                case "min" -> delegateFieldType(Metric.min).blockLoader(blContext);
+                case "sum" -> delegateFieldType(Metric.sum).blockLoader(blContext);
+                case "count" -> {
+                    var fieldType = delegateFieldType(Metric.value_count);
```

Comment:
This is one of the hacks. This makes sure to load double blocks, otherwise a class cast exception occurs in `SumDoubleAggregatorFunction.java:60`